server/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -245,11 +268,15 @@ public void afterRefresh(boolean didRefresh) throws IOException {
         // reopen, and so any concurrent indexing requests can still sneak in a few additions to that current map that are in fact
         // reflected in the previous reader. We don't touch tombstones here: they expire on their own index.gc_deletes timeframe:
 
-        maps = maps.invalidateOldMap();
-        assert (unsafeKeysMap = unsafeKeysMap.invalidateOldMap()) != null;
+        maps = maps.invalidateOldMap(archiver);
+        assert (unsafeKeysMap = unsafeKeysMap.invalidateOldMapForAssert()) != null;
 
     }
 
+    public void afterUnpromotablesRefreshed(long generation) {
+        archiver.afterUnpromotablesRefreshed(generation);
+    }
+
```

Comment:
I think this method can move to the plugin? It is not called in server and seems to be a concern only for the plugin.